common/lib/xmodule/xmodule/tests/test_library_content.py

```diff
@@ -140,3 +163,62 @@ def test_validation(self):
         # Now if we update the block, all validation should pass:
         self.lc_block.refresh_children(None, None)
         self.assertTrue(self.lc_block.validate())
+
+        # Set max_count to higher value than exists in library
+        self.lc_block.max_count = 50
+        result = self.lc_block.validate()
+        self.assertFalse(result)  # Validation fails due to at least one warning/message
+        self.assertTrue(result.summary)
+        self.assertEqual(StudioValidationMessage.WARNING, result.summary.type)
+        self.assertIn("only 4 matching problems", result.summary.text)
+
+        # Add some capa problems so we can check problem type validation messages
+        self.lc_block.max_count = 1
+        self._create_capa_problems()
+        self.lc_block.refresh_children(None, None)
+        self.assertTrue(self.lc_block.validate())
+
+        # Existing problem type should pass validation
+        self.lc_block.max_count = 1
+        self.lc_block.capa_type = 'multiplechoiceresponse'
+        self.assertTrue(self.lc_block.validate())
+
+        # ... unless requested more blocks than exists in library
+        self.lc_block.max_count = 3
+        self.lc_block.capa_type = 'multiplechoiceresponse'
+        result = self.lc_block.validate()
+        self.assertFalse(result)  # Validation fails due to at least one warning/message
+        self.assertTrue(result.summary)
+        self.assertEqual(StudioValidationMessage.WARNING, result.summary.type)
+        self.assertIn("only 1 matching problem", result.summary.text)
+
+        # Missing problem type should always fail validation
+        self.lc_block.max_count = 1
+        self.lc_block.capa_type = 'customresponse'
+        result = self.lc_block.validate()
+        self.assertFalse(result)  # Validation fails due to at least one warning/message
+        self.assertTrue(result.summary)
+        self.assertEqual(StudioValidationMessage.WARNING, result.summary.type)
+        self.assertIn("no matching problem types", result.summary.text)
+
+    def test_capa_type_filtering(self):
+        """
+        Test that the capa type filter is actually filtering children
+        """
+        self._create_capa_problems()
```

Comment:
Test the case of a problem having more than one type (for example, CAPA problem with both multiple choice and select inputs)?